wagtail/contrib/forms/utils.py

```diff
@@ -16,19 +16,13 @@ def get_field_clean_name(label):
     return safe_snake_case(label)
 
 
+@lru_cache(maxsize=None)
 def get_form_types():
```

Comment:
I initially thought the issue was because of the FTS behaviour of MySQL/MariaDB. I had a commit where I changed the tests to use `TransactionTestCase`, but then I got a strange error where the listing will never show any pages after the first test method. Turns out, it's because the `ContentType` table also gets truncated during teardown, and they get repopulated with different IDs.

I needed to find a way to clear the cached form types, so I refactored this function to use `lru_cache` instead. This is no longer needed now that I've found the true cause of the failing test, but I think this cleanup is still worthwhile.